### PR TITLE
refactor: Align overstep limit

### PR DESCRIPTION
**Pull request by andiwand**
Overstep limits are defined differently in our 3 steppers. This PR aligns them to the convention in the `EigenStepper`. The `StraightLineStepper` returned the wrong sign which might have affected Fatras simulation for neutral particles.

Pulled out of https://github.com/acts-project/acts/pull/2336

### CODE REVIEW

File: Core/include/Acts/Propagator/StraightLineStepper.hpp [hunk 1]
```diff
@@ -222,7 +222,7 @@ class StraightLineStepper {
   /// @param state The stepping state (thread-local cache)
   double overstepLimit(const State& state) const {
     (void)state;
```
Comment (benjaminhuth) [inline Core/include/Acts/Propagator/StraightLineStepper.hpp:224]:
```suggestion
  double overstepLimit(const State& /*state*/) const {
```

**Comment by andiwand** [comment]
I looked at the tracksummary and it seems like we reconstruct 4 additional tracks which have some holes and bad pT estimate. I do not think existing performance is impacted but I could double check if necessary

Opinion @paulgessinger @benjaminhuth ?

**Comment by benjaminhuth** [comment]
> I looked at the tracksummary and it seems like we reconstruct 4 additional tracks which have some holes and bad pT estimate. I do not think existing performance is impacted but I could double check if necessary

This means, the fitted params in the tracksummary of particles that where reconstructed before don't change significantly, right? 
So we can trace the changes back to diff in simulation, that allows more particles to pass the truth seeding cuts or so?

**Comment by andiwand** [comment]
> This means, the fitted params in the tracksummary of particles that were reconstructed before don't change significantly, right? So we can trace the changes back to diff in simulation, that allows more particles to pass the truth seeding cuts or so?

This or it was a fit with errors before.

I guess I could also try to diff the particles/hits to see if my hypothesis holds. But I guess a bunch of them also have some slight numerical differences

**Comment by paulgessinger** [comment]
The ttbar job is pretty low-stats in any case, so not sure how much we need to read into fluctuations there.

**Comment by andiwand** [comment]
physmon diff seems to be down to https://github.com/acts-project/acts/pull/2441 and CKF diff with truth smeared seeds. To me it looks like we pick up more surfaces which is coming from a diff in the navigation but it is hard to tell which one is correct. Since it is only happening for truth smeared it could point to an existing/fixed navigation issue before the first sensitive surfaces because the other seeders would start directly from that surface

**Review by benjaminhuth — DISMISSED** [review]

File: Core/include/Acts/Propagator/StraightLineStepper.hpp [hunk 2]
```diff
@@ -220,9 +220,8 @@ class StraightLineStepper {
   /// Overstep limit
   ///
   /// @param state The stepping state (thread-local cache)
```
Comment (andiwand) [inline Core/include/Acts/Propagator/StraightLineStepper.hpp:222]:
```suggestion
```

**Review by benjaminhuth — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456987) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456986) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456985) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456984) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456983) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456982) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456981) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456980) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456979) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456978) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456977) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456976) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456975) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456974) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456973) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456972) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/32456971) |

**Pull request merged**